tensorflow/compiler/xla/stream_executor/dnn.h

```diff
@@ -152,6 +155,75 @@ enum class RnnDirectionMode {
 // DepthToSpace comment for more information.
 enum class DepthToSpaceLayout { DepthHeightWidth };
 
+class TensorDescriptor {
```

Comment:
This is an almost exact clone of xla::Shape, right? I wonder if it's possible to unify those better, I guess lack of SE->XLA dependencies prevents it, but then do we need a whole new abstraction there?

Reply:
> This is an almost exact clone of xla::Shape, right? I wonder if it's possible to unify those better, I guess lack of SE->XLA dependencies prevents it, but then do we need a whole new abstraction there?

You are right..the information here can be extracted form xla::Shape but as you said there is no way to have that dependency and reuse xal::Shape here. However, this is much more light weight than `Shape` so it's not too bad. I looked to see if there is anything that already exists and fulfills this requirement but couldn't find any that can be reused.